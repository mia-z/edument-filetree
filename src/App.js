import React, { useReducer } from 'react';
import './App.scss';
import "bootstrap/dist/css/bootstrap.min.css";
import { HashRouter as Router } from "react-router-dom";
import { Container } from "react-bootstrap";
import DirectoryView from "./Components/DirectoryView";
import FilePath from "./Components/SubComponents/FilePath";
import { DirectoryReducer, initialState } from "./Reducers/DirectoryReducer";
import BoxView from "./Components/BoxView";
import MenuBar from "./Components/MenuBar";

function App() {
    const [state, dispatch] = useReducer(DirectoryReducer, initialState);

    return (
        <Router>
            <Container>
                <BoxView>
                    <MenuBar DirectoryState={state} dispatch={dispatch} />
                    <FilePath DirectoryState={state} dispatch={dispatch} />
                    <DirectoryView DirectoryState={state} dispatch={dispatch} />
                </BoxView>
            </Container>
        </Router>
    );
}

export default App;
