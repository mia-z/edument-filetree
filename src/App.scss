@import "scss/vars";

html, div#root {
    background-color: $secondhand-grey;
}
