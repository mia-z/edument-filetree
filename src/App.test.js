import React from 'react';
import { create } from 'react-test-renderer';
import App from './App';

test('Makes sure the root component renders without crashing', () => {
  const app = () => create(<App />);
  expect(app).toMatchSnapshot(App);
});
