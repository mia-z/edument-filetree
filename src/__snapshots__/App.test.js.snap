// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Makes sure the root component renders without crashing 1`] = `[Function]`;
